### PR TITLE
Add CPU affinity for PoH service thread

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
The PoH Service thread runs in a tight loop to run PoH. It can benefit from CPU affinity.

#### Summary of Changes
Added CPU affinity

### CODE REVIEW

File: core/src/poh_service.rs [hunk 1]
```diff
@@ -27,12 +28,16 @@ impl PohService {
     ) -> Self {
         let poh_exit_ = poh_exit.clone();
         let poh_config = poh_config.clone();
+        let core_ids = core_affinity::get_core_ids();
```
Comment (mvines) [inline core/src/poh_service.rs]: any reason to not move this line right above line 38?

Comment (pgarg66) [reply core/src/poh_service.rs]: no reasons

File: core/src/poh_service.rs [hunk 2]
```diff
@@ -33,6 +34,9 @@ impl PohService {
                 if poh_config.hashes_per_tick.is_none() {
                     Self::sleepy_tick_producer(poh_recorder, &poh_config, &poh_exit_);
                 } else {
+                    if let Some(cores) = core_affinity::get_core_ids() {
```
Comment (garious) [inline core/src/poh_service.rs]: @pgarg66 It's important that you comment the hell out of side-effects like this that can't be unit-tested. What would discourage someone from deleting this code?

Comment (pgarg66) [reply core/src/poh_service.rs]: @garious , sure, I'll add some comments in the next PoH related PR (coming soon)

Comment (pgarg66) [reply core/src/poh_service.rs]: @garious this PR has the comment: https://github.com/solana-labs/solana/pull/4396

**Comment by pgarg66** [comment]
https://github.com/solana-labs/solana/issues/4331

**Pull request merged**